nautobot/extras/models/groups.py

```diff
@@ -56,30 +56,198 @@ class DynamicGroup(OrganizationalModel):
+            # only being done for CharField since in the filterset form this ends up being a
+            # `MultVarCharField` (dynamically generated from from `MultiValueCharFilter`) which is
+            # not correct for char fields.
+            if isinstance(modelform_field, forms.CharField):
+                # Get ready to replace the form field w/ correct widget.
+                new_modelform_field = filterset_field.field
+                new_modelform_field.widget = modelform_field.widget
+
+                # If `required=True` was set on the model field, pop "required" from the widget
+                # attributes. Filter fields should never be required!
+                if modelform_field.required:
+                    new_modelform_field.widget.attrs.pop("required")
+
+                modelform_field = new_modelform_field
+
+            # Carry over the `to_field_name` to the modelform_field.
+            to_field_name = filterset_field.extra.get("to_field_name")
+            if to_field_name is not None:
+                modelform_field.to_field_name = to_field_name
+
+            logger.debug("Added %s (%s) to filter fields", missing_field, modelform_field.__class__.__name__)
+            filterform_fields[missing_field] = modelform_field
+
+        # Reduce down to a final dict of desired fields.
+        return_fields = {}
+        for field_name, filter_field in filterform_fields.items():
+            # Skip excluded fields
+            if field_name.startswith(self.exclude_filter_fields):
+                logger.debug("Skipping excluded filter field: %s", field_name)
+                continue
+
+            return_fields[field_name] = filter_field
+
+        return return_fields
+
+    def get_filter_fields(self):
+        """Return a mapping of `{field_name: filter_field}` for this group's `content_type`."""
+        # Fail cleaninly until the object has been created.
+        if self.model is None:
+            return {}
+
+        if not self.present_in_database:
+            return {}
+
+        return self._map_filter_fields()
+
+    def get_queryset(self, flat=False):
+        """
+        Return a queryset for the `content_type` model of thhis group.
+
+        The queryset is generated based on the `filterset_class` for the Model.
+        """
+
+        model = self.model
 
         if not self.filter:
             return model.objects.none()
 
-        qs = self.map.get_queryset(self.filter, **kwargs)
+        filterset = self.filterset_class(self.filter, model.objects.all())
+        qs = filterset.qs
 
-        # Make sure that this instance can't be a member of its own group
+        # Make sure that this instance can't be a member of its own group.
         if self.present_in_database and model == self.__class__:
             qs = qs.exclude(pk=self.pk)
 
+        # If `flat` is desired, return a flat list of "pk".
+        if flat:
+            qs = qs.values_list("pk", flat=True)
+
```

Comment:
Wondering if we even need this here? Looking at this PR and #1425, it seems like `flat=True` is only used once in `DynamicGroupQuerySet.get_for_object()`, and it would probably be clearer there to change:

```python
            if obj.pk in dynamic_group.get_queryset(flat=True):
```

to

```python
            if obj.pk in dynamic_group.get_queryset().values_list("pk", flat=True):
```